litex/soc/interconnect/csr.py

```diff
@@ -268,10 +268,17 @@ def read(self):
 
     def write(self, value):
         """Write method for simulation."""
-        yield self.storage.eq(value >> self.alignment_bits)
-        yield self.re.eq(1)
+        for sc in reversed(self.simple_csrs):
+            yield from sc.write(value)
+            yield
+            value = value >> sc.size
+
+    def write_from_dev(self, value):
+        yield self.dat_w.eq(value)
+        yield self.we.eq(1)
         yield
-        yield self.re.eq(0)
+        yield # FIXME: Why two yields!?
```

Comment:
Couldn't figure out why two yields here...

Reply:
It seems to me that having any `yield` statement in simulation code could be a problem.  For example, if you wanted to update two different CSRs in the same clock cycle, if they both did this then your simulation would go forward several cycles.

Or is this the reason why you oversample the clock by 10x?